### ISSUE TITLE
Using virtual display causes stutter or freeze every 10 seconds (exactly)

### ISSUE THREAD

**Issue by Fel-the-Insane** [ClassicOldSong/Apollo#179]
When using virtual display the stream will stutter/lag/freeze every 10 seconds, can count it out to the second every time. 

This issue happens regardless of if I use moonlight or artemis client. It happens on both android and Xbox. 

The only fix is to not use virtual display and to instead manually change resolution. 

Notably I set up apollo on my sons gaming PC exactly the same and he has no issues. 

My system has ryzen 9 7950x and rtx4070, my sons is a prebuilt INTEL with a 3060ti.

**Comment by ClassicOldSong** [comment]
Try set Adapter Name to your RTX 4090 and enable headless mode.

**Comment by Fel-the-Insane** [comment]
Still happening with those changes unfortunately.

**Comment by ClassicOldSong** [comment]
hmmmm

Can you try disconnect your monitor physically and stream again?

**Comment by ClassicOldSong** [comment]
On Android, download the latest pre-release and choose Warp Drive in frame pacing might help. On other platforms there's no luck... I'm unsure about why this happens but my gut tells me it's a problem burying deep in the GPU core that can only be fixed by Nvidia. The thing is, when using a physical monitor, the clock signals is fetched from the monitor itself, but with a virtual monitor it's generated internally by your GPU or CPU. There's a mismatch between the two clocks and the hardware encoder gets confused. Your case might be an amplified version of what I tried to fix in the last build, but I failed because I then realized that it might be a hardware problem.

Some more things to try can be:

1. Disable the physical monitor in Windows settings while streaming and see if it makes a difference 
2. Make sure the Adapter Name is set correctly: you can copy the name of your desired GPU for encoding from the results of dxgi-info.exe
3. Use a different monitor
4. Use headless mode and set Adapter Name to your iGPU, in this case it'll use your iGPU to encode which might workaround this problem, but it occupies PCIE bandwidth that might reduce some gaming performance. The good thing with this method is it actually frees computing power from your gaming GPU, so the results might vary.

**Comment by Fel-the-Insane** [comment]
I think you're probably right about it being an NVIDIA thing and I appreciate your time helping me.  As for the troubleshooting you suggested; nothing seemed to change anything except using a different monitor, and upon further investigation it seems that was only because I was using an HDMI cable for the other monitor. Having a DisplayPort used regardless of which port (iGPU or any of the 3 on graphics card) immediately makes the issue return.

So my final solution might just be getting a dummy plug and seeing if that has any different results than virtual display as I'm not willing to sacrifice my refresh rate by using HDMI on my monitor. I'll still use Apollo as well cuz you fixed that horrible issue on XBOX.

**Comment by ClassicOldSong** [comment]
It's not me fixed the xbox issue, it's @TheElixZammuto . Now the change made by him has merged to Sunshine, but it's in the pre-release only.

**Comment by Fel-the-Insane** [comment]
Well thank you to them as well.

**Comment by BillboardEric** [comment]
@Fel-the-Insane Any chance you found a workaround? Experiencing the exact issue.

**Comment by Fel-the-Insane** [comment]
> [@Fel-the-Insane](https://github.com/Fel-the-Insane) Any chance you found a workaround? Experiencing this exact issue.

No, I just can't use the virtual display with my setup. The automatic resolution switching still works without that though so its fine for now. If I had an OLED I would have went the dummy plug route instead to avoid weird burn in.
